<launch>

    <rosparam command="load" file="$(find hebi_rosie_demo)/parameters/arm.txt" />
    <rosparam command="load" file="$(find hebi_rosie_demo)/parameters/modules.txt" />
    <rosparam command="load" file="$(find hebi_rosie_demo)/parameters/colors.txt" />
    <rosparam command="load" file="$(find hebi_rosie_demo)/parameters/calibration.txt" />

    <include file="$(find realsense2_camera)/launch/rs_camera.launch" />
    <node pkg="hebi_rosie_demo" type="vision_process" name="vision_process" />

    <group ns="rosie">
        <node ns="base" pkg="hebi_cpp_api_examples" type="omni_base_node" name="omni_base_node" />
        <node ns="arm" pkg="hebi_cpp_api_examples" type="arm_node" name="arm_node_action" />
        <group ns="gripper">
            <rosparam command="load" file="$(find hebi_cpp_api_examples)/config/parallel-gripper_params.yaml" />
            <param name="family" value="Rosie" />
            <node pkg="hebi_cpp_api_examples" type="gripper_node" name="gripper_node" />
            <node pkg="hebi_rosie_demo" type="gripper_service_node.py" name="gripper_service_node" />
        </group>
    </group>

    <param name="rosie/arm/ik_seed" value="[0.0, 0.7854, 1.57075, 0.7854, -3.1415, 1.57075]" />
    <node pkg="hebi_rosie_demo" type="demo_central" name="demo_central" output="screen" required="true" />

</launch>
